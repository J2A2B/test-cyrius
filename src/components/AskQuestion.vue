<template>
  <div>
    <h2 id="question-text">{{ question.text }}</h2>
    <button id="select-true-answer" @click="respondAnAnswer(true)">
      {{ WORDINGPAGE.true }}
    </button>
    <button id="select-false-answer" @click="respondAnAnswer(false)">
      {{ WORDINGPAGE.false }}
    </button>
  </div>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits, PropType } from "vue";

interface Question {
  text: string;
  answer: boolean | null;
}
interface WordingPage {
  true: string;
  false: string;
}

const WORDINGPAGE: WordingPage = {
  true: "Vrai",
  false: "Faux",
};
const props = defineProps({
  question: { type: Object as PropType<Question>, default: () => ({}) },
  index: { type: Number },
});

const emit = defineEmits(["emitAnswer"]);

const respondAnAnswer = (value: boolean) => {
  emit("emitAnswer", { index: props.index, answer: value });
};
</script>

<style scoped>
button {
  margin: 10px;
  width: 80px;
  padding: 5px;
  border: 2px solid grey;
  border-radius: 3px;
  color: white;
  background-color: black;
}
</style>
