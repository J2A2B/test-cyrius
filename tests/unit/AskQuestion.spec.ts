import { shallowMount, mount } from "@vue/test-utils";
import AskQuestion from "@/components/AskQuestion.vue";

describe("QuizTest.vue", () => {
  describe("Template", () => {
    it("should match snapshot", async () => {
      const wrapper = shallowMount(AskQuestion);

      expect(wrapper.element).toMatchSnapshot();
    });
  });
  describe("Props", () => {
    describe("When the props question is filled", () => {
      it("should display the title of the question", async () => {
        const question = {
          text: "Question title",
          answer: true,
          correctAnswer: true,
        };
        const wrapper = mount(AskQuestion, {
          props: {
            question,
          },
        });
        expect(wrapper.find("h2#question-text").text()).toEqual(question.text);
      });
    });
  });
  describe("events", () => {
    describe("on click", () => {
      it("should emit click with a true answer", () => {
        const question = {
          text: "Question title",
          answer: true,
          correctAnswer: true,
        };
        const wrapper = mount(AskQuestion, {
          props: { question, index: 1 },
        });

        wrapper.vm.respondAnAnswer(true);

        const expectedEmittedValue = {
          answer: true,
          index: 1,
        };
        expect(wrapper.emitted().emitAnswer[0]).toEqual([expectedEmittedValue]);
      });
      it("should emit click with a false answer", () => {
        const question = {
          text: "Question title",
          answer: false,
          correctAnswer: true,
        };
        const wrapper = mount(AskQuestion, {
          props: { question, index: 1 },
        });

        wrapper.vm.respondAnAnswer(false);

        const expectedEmittedValue = {
          answer: false,
          index: 1,
        };
        expect(wrapper.emitted().emitAnswer[0]).toEqual([expectedEmittedValue]);
      });
    });
  });
});
