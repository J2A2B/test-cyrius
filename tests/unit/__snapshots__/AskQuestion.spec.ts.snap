// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizTest.vue Template should match snapshot 1`] = `
<div>
  <h2
    id="question-text"
  />
  <button
    id="select-true-answer"
  >
    Vrai
  </button>
  <button
    id="select-false-answer"
  >
    Faux
  </button>
</div>
`;
